### PR TITLE
add inner_id support for map of map

### LINKED ISSUE #618: Support for id / inner_id for map of maps
When there are many maps defined in an ebpf program (and also in different files), it is difficult to figure out the "inner_map_idx" for the inner maps (which is the offset of the inner map in the `maps` section of ELF file) beforehand. This is also not scalable as adding a new map in the middle of the maps section later breaks the "inner_map_idx" assumption.

We should add support for "id" and "inner_id" there the inner map defines an ID, and the outer map has "inner_id" se to the ID of the inner map. This uniquely identifies the inner map template in ELF file irrespective of the map order.

A reference:
https://github.com/iovisor/gobpf/issues/120

### PR DESCRIPTION
**Pull request by saxena-anurag**
Fixes #618 

The fix first tries to find inner map based on `inner_map_idx`. If `inner_map_idx` is 0, it tries to use `inner_id` to find the inner map template.

### CODE REVIEW

File: include/ebpf_structs.h [hunk 1]
```diff
@@ -69,6 +69,11 @@ typedef struct _ebpf_map_definition_in_file
      * file is the inner map template.
      */
     uint32_t inner_map_idx;
+
+    // id is the identifier for a map template.
```
Comment (dthaler) [inline include/ebpf_structs.h:73]: I don't think this comment style will show up in doxygen docs like lines 61-65 do

**Review by dthaler — APPROVED** [review]
Ok except please fix comment style in ebpf_struct.h

File: include/ebpf_structs.h (hunk 1 above)
Comment (saxena-anurag) [reply include/ebpf_structs.h]: fixed.

File: tests/sample/map_in_map_v2.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(uint32_t),
+    .value_size = sizeof(uint32_t),
+    .max_entries = 1,
+    .id = INNER_MAP_ID};
+
+SEC("xdp_prog") int lookup(struct xdp_md* ctx)
+{
+    uint32_t outer_key = 0;
+    void* nolocal_lru_map = bpf_map_lookup_elem(&outer_map, &outer_key);
```
Comment (shankarseal) [inline tests/sample/map_in_map_v2.c:28]: why use lru here ? Neither of the map types are LRU. Or is lru used in some other context?

**Review by shankarseal — APPROVED** [review]

File: tests/sample/map_in_map_v2.c (hunk 2 above)
Comment (saxena-anurag) [reply tests/sample/map_in_map_v2.c:28]: This is just wrong naming of variables. Will fix it.

Comment (saxena-anurag) [reply tests/sample/map_in_map_v2.c]: Fixed.

**Pull request merged**